unit_tests/sources/streams/http/test_http.py

```diff
@@ -331,6 +336,34 @@ def test_raise_on_http_errors(mocker, error):
     assert send_mock.call_count == stream.max_retries + 1
 
 
+class StubHttpStreamWithErrorHandler(StubBasicReadHttpStream):
+    def get_error_handler(self) -> Optional[ErrorHandler]:
+        return HttpStatusErrorHandler(logging.getLogger())
+
+
+def test_dns_resolution_error_retry():
+    """Test that DNS resolution errors are retried"""
+    stream = StubHttpStreamWithErrorHandler()
+    error_handler = stream.get_error_handler()
+    request = requests.PreparedRequest()
+    request.url = "https://example.com"
+    dns_error = DNSResolutionError(
```

Comment:
Can you add a test that simulates/mocks the error we expect the http api to raise? Specifically, I think we're expecting something like an invalid URL code, except with the noted error text about DNS.